File: MB-Whatsapp/providers_py/gemini/adapter.py
from __future__ import annotations

from ..provider_health import disabled_provider_health
from ..provider_response import build_provider_response
from ..types import ProviderAdapter, ProviderCapabilities, ProviderRequest, ProviderResponse


class DisabledGeminiAdapter(ProviderAdapter):
    def __init__(self) -> None:
        super().__init__(
            provider_name="gemini",
            model_name="gemini-disabled",
            capabilities=ProviderCapabilities(True, False, False, True, False, False, 8192, False),
        )

    def complete(self, request: ProviderRequest) -> ProviderResponse:
        return build_provider_response(request, provider_name="gemini", model_name=self.model_name, status="disabled", safety_flags=["NO_EXTERNAL_CALL"])

    def healthcheck(self):
        return disabled_provider_health("gemini")

File: MB-Chat/providers/gemini/adapter.ts
import { buildProviderResponse } from "../provider-response";
import type { ProviderRequest, ProviderResponse } from "../types";
import { GEMINI_CAPABILITIES } from "./capabilities";
import { geminiHealthDisabled } from "./health";
import type { GeminiProviderAdapter } from "./contract";

export class DisabledGeminiAdapter implements GeminiProviderAdapter {
  readonly provider_name = "gemini" as const;
  readonly model_name = "gemini-disabled";
  readonly capabilities = GEMINI_CAPABILITIES;

  async complete(input: ProviderRequest): Promise<ProviderResponse> {
    return buildProviderResponse({
      request: input,
      provider_name: "gemini",
      model_name: this.model_name,
      status: "disabled",
      safety_flags: ["PROVIDER_DISABLED", "NO_EXTERNAL_CALL"],
    });
  }

  async healthcheck() {
    return geminiHealthDisabled();
  }
}
